tools/config_gen.py

```diff
@@ -170,7 +171,10 @@ def pull_base_toml() -> typing.Dict:
 
     # Now, we need to replace any placeholders that might be present in the base toml file with the real verisons.
     user = getpass.getuser()
-    pwd = str(repo_path.resolve())
+    if "IN_NIX_SHELL" in os.environ and "src" in os.environ:
+        pwd = os.environ['src']
```

Comment:
Who sets the `src` env var?

Reply:
Nix. I was surprised it isn't called `NIX_SRC_DIR` or something. I'm using this one so the placeholders get replaced with `/nix/store` store paths, rather than paths into your repo. This makes the output stable and also if the config file is ever a Nix output it'll have the correct dependencies.